### PR TITLE
Update 'list'

### LINKED ISSUE #25: Refactor list functionality
'list' portion under the `receiveInput()` function can be refactored to the same level of abstraction as the other portion in the same function.

### PR DESCRIPTION
**Pull request by JuneNita**
* list by keyword (name, ingredients), and filtered by keywords (using &)
* add listRecipeTest
* update user guide for 'list'
* closes #25

### CODE REVIEW

File: src/main/java/seedu/meal360/Meal360.java [hunk 1]
```diff
@@ -39,7 +39,8 @@ public static void receiveInput(String input) {
             Recipe recipe = parser.parseViewRecipe(command, recipeList);
             ui.printRecipe(recipe);
         } else if (command[0].equals("list")) {
-            parser.parseListRecipe(recipeList);
+            RecipeList recipeListToPrint = parser.parseListRecipe(command, recipeList);
+            ui.listRecipe(recipeListToPrint);
```
Comment (jaredoong) [inline src/main/java/seedu/meal360/Meal360.java:43]: Good abstraction!

**Review by jaredoong — APPROVED** [review]
Neat code with sufficient abstraction. LGTM

File: src/main/java/seedu/meal360/RecipeList.java [hunk 2]
```diff
@@ -29,23 +28,20 @@ public Recipe deleteRecipe(int recipeNum) {
...
+        RecipeList filteredRecipeList = new RecipeList();
+        if (filters == null) {
+            return this;
+        }
+        for (Recipe recipe: this) {
+            filteredRecipeList.add(recipe);
+            for (String filter : filters) {
+                filter = filter.trim();
+                if (!recipe.getName().contains(filter) && !recipe.getIngredients().containsKey(filter)) {
+                    filteredRecipeList.remove(recipe);
```
Comment (TopGun2001) [inline src/main/java/seedu/meal360/RecipeList.java:41]: I liked the way how you used `for-each loop` to iterate through.

File: src/test/java/seedu/meal360/Meal360Test.java [hunk 3]
```diff
@@ -44,4 +46,45 @@ public void testDeleteRecipe() {
         // check if recipe list size is 1 less than before
         assertTrue((oldRecipeListSize - 1) == newRecipeListSize);
     }
+
+    @Test
+    public void testListRecipe() {
```
Comment (TopGun2001) [inline src/test/java/seedu/meal360/Meal360Test.java:51]: Good enough test conditions for junit.

**Pull request merged**